src/Cassandra/Requests/RequestExecution.cs

```diff
@@ -389,7 +396,7 @@ private void HandleRequestError(IRequestError error, Host host)
                     break;
             }
 
-            _requestObserver.OnRequestError(host, retryInformation.Reason, retryInformation.Decision.DecisionType);
+            _requestObserver.OnNodeRequestError(host, retryInformation.Reason, retryInformation.Decision.DecisionType, _requestTrackingInfo, ex);
```

Comment:
Hmm we might need to ensure this gets called before `_parent.SetCompleted` because otherwise the "OnRequestFailure" will get called before "OnNodeRequestError" on the request tracker...

And if the retry decision is `Retry` then the new "child trace" will be started before this one ends.

```
switch (retryInformation.Decision.DecisionType)
    {
        case RetryDecision.RetryDecisionType.Rethrow:
           _requestObserver.OnNodeRequestError(host, retryInformation.Reason, RetryDecision.RetryDecisionType.Rethrow, _requestTrackingInfo, ex);
            _parent.SetCompleted(ex);
            break;
        case RetryDecision.RetryDecisionType.Ignore:
            _requestObserver.OnNodeRequestError(host, retryInformation.Reason, RetryDecision.RetryDecisionType.Ignore, _requestTrackingInfo, ex);
            // The error was ignored by the RetryPolicy, return an empty rowset
            _parent.SetCompleted(null, FillRowSet(RowSet.Empty(), null));
            break;
        case RetryDecision.RetryDecisionType.Retry:
            _requestObserver.OnNodeRequestError(host, retryInformation.Reason, RetryDecision.RetryDecisionType.Retry, _requestTrackingInfo, ex);
            //Retry the Request using the new consistency level
            Retry(retryInformation.Decision.RetryConsistencyLevel, retryInformation.Decision.UseCurrentHost, host);
            break;
        default: 
           _requestObserver.OnNodeRequestError(host, retryInformation.Reason, retryInformation.Decision.DecisionType, _requestTrackingInfo, ex);
    }
```

The above isn't pretty but I don't think there's pretty and efficient solutions for this

Reply:
Changed it in the last commit. Please review